### DISCUSSION TITLE
Privacy and Data Control for Filters in User Profiles

### DISCUSSION THREAD

**Discussion by ajbott** [PullDating/Org#209] [Ideas & Features]
## 1. Feature Overview
The **Privacy Filters** feature enables users to control the visibility of their personal information on their dating profile. Users can mark each piece of information, or 'filter', as 'Public' or 'Private'. Public filters are visible to all app users, while private filters are used in the backend for matchmaking purposes but not displayed on the user's profile.

## 2. Purpose
This feature is designed to empower users with greater control over their personal data, thereby enhancing trust and privacy within the app. It allows users to manage the visibility of their profile information, supporting more meaningful connections while respecting individual privacy preferences.

### Setting Privacy Levels
- Users can select 'Public' to make a filter visible on their profile or 'Private' to keep it hidden but still use it for matchmaking.

### User Interface
- Privacy settings will be easily accessible next to each filter in the user's profile, using clear indicators such as toggle switches or dropdown menus.

### Backend Integration
- Filters marked as 'Private' will inform the app's matchmaking algorithm without being visible on the user's public profile.

### User Education and Transparency
- The app will provide clear information on how privacy settings affect profile visibility and matchmaking, ensuring users understand the implications of their choices.

### Privacy Settings Management
- Users will have the flexibility to review and adjust their privacy settings at any time.

## 4. Technical Considerations
- Implementation will require updates to the user interface, data handling, and matchmaking algorithms.
- Privacy settings need to be stored securely and applied consistently.

## 5. Expected Impact
- Enhanced user trust and satisfaction by offering control over personal information.
- Improved match quality by utilizing a comprehensive set of user preferences in the matchmaking process, including those not publicly displayed.

## Conclusion
The **Customizable Privacy Filters** feature represents a significant step forward in user-centric design within dating apps, prioritizing user privacy and control.

**Comment by Oshyan** [comment]
If I've understood what is being proposed here, then I have a concern with conflating profile information with "filters". I think in this case it may be based on the assumption that someone's profile info/preference directly matches what they are looking for, but I don't think this is a reasonable assumption. Perhaps I've misunderstood, if so please let me know.

Assuming I'm understanding correctly, then I would break it down like this:
- Profile info
  - Public - a certain set of profile fields should be mandatory and public
  - Private - e.g. you may not want to answer what school you went to or a specific kink publicly, but you can answer and still match with people who are filtering on that stuff; there are already dating apps that allow this, like OKC
- "Looking For" - A more static representation of what someone is looking for, which becomes information that is (in some cases optionally) displayed on a person's profile and is a representation of certain of the other user's filters, but may not be ideally called "Filters" on a user's profile since this risks being confused with filtering that a user can do themselves on what profiles *they* see
  - Public - "This person is looking for X, Y, Z" (e.g. man, casual encounters, etc.)
  - Private - A person's "looking for" (which again *may not necessarily match their own profile characteristics) which they might choose to hide for some reason, e.g. kinks, etc.
- Filters/Filtering - An action and set of options each user is provided to limit what profiles *they* see and can interact with
  - Mandatory "filters" - there should be a set of the "filters" or "looking for" that is mandatory for basic matching, stuff like what gender(s)/sex(es) they are looking for, what kind of relationship(s), etc.
  - Optional filters - things like height, interests, etc.

Hopefully that is clarifying. The main points I'm trying to make are:
1. What someone is looking for is *not* necessarily the same as their own profile characteristics
2. Profile characteristics, or indications of what a user is looking for, should not be called "filters" to other users viewing that profile
3. We should carefully consider what information or "filters" are mandatory and which are not
4. Whether a "filter" is visible or private should be separate from all that, but some may be mandatorily displayed on profiles

In general I know there is a lot of intention to provide more "private" options and functionality, and I am in support of that overall intention. However I think it can very easily and quickly complicate and compromise the overall user experience and community health if privacy is prioritized above all else. Online dating inevitably involves some level of compromise, and the community feeling, etc. is affected by choices around privacy, anonymity, etc. It is also a minority of users who have a high concern for privacy (the remaining popularity of very not-private social media and services demonstrates this), and as a general policy I do not think we should compromise the experience of the majority for the needs of any minority. 

The needs of minorities should **absolutely** be understood and addressed as much as possible, but they *must* be balanced with the overall health of the community and needs of the average user. I think there are ways of doing this effectively, and I continue to fall back on the idea of incentivizing but not requiring behavior and choices that benefit the overall user base. E.g. a user might choose to make most or all of their "filters" private on their profile, but if this is a concern for *other* users then they should be empowered to filter *out* people with a ton of "private" information. And in that case the "private" user should be made aware that turning on "private everything" will limit their visibility. Maybe that's an outcome they want, or maybe they'll reconsider their choice and balance it with the benefits of being seen and searchable by more people who explicitly *want* to see as much info about their prospective matches as possible. 

One way to implement this in a simple way without having to provide a system to filter based on whether other users have marked specific fields as "private" could be:
1. Allow marking certain fields as private, some of which may be mandatory ones
2. Allow other users to filter who *they* see based on two criteria:
a. Whether a user has hidden mandatory fields or only additional/advanced/extended ones
b. A rough calculated average of the amount of hidden information on a given user's profile, e.g. when filtering for profiles to browse, a user would be able to filter out people whose "privacy" is "low, medium, high", something like that
